src/content/docs/en/guides/backend/supabase.mdx

```diff
@@ -3,10 +3,806 @@ title: Supabase & Astro
+
+- `@supabase/supabase-js`
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+    ```shell
+    npm install @supabase/supabase-js
+    ```
+  </Fragment>
+  <Fragment slot="pnpm">
+    ```shell
+    pnpm install @supabase/supabase-js
+    ```
+  </Fragment>
+  <Fragment slot="yarn">
+    ```shell
+    yarn add @supabase/supabase-js
+    ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Create Profile Table
+
+1. Go to [Studio URL](http://localhost:54323)
+2. Click on SQL Editor on the left sidebar
+3. Choose `User Management Starter` from `Quick start` in the bottom of the page
+4. Click `RUN`
+
+### Create Supabase Client
+
+1. Create a directory named `lib` in the `src/` directory.
+2. Create a file named `supabaseClient.ts` in the `src/lib/` directory.
+
+In `supabaseClient.ts` add the following code to create the supabase client.
+
+```ts title="src/lib/supabaseClient.ts"
+import { createClient } from '@supabase/supabase-js'
+
+export const supabase = createClient(
+  `${import.meta.env.SUPABASE_URL}`,
+  `${import.meta.env.SUPABASE_ANON_KEY}`,
+  {         
+    auth: {
+      persistSession: false,
+    }
+  }
+)
+```
+
+The project structure should include these new files:
+
+<FileTree title="Project Structure">
+- ...
+- src/
+  - lib/
+    - supabaseClient.ts
+  - env.d.ts
+- .env
+- astro.config.mjs
+- package.json
+- ...
+</FileTree>
+
+### Prepare Authentication Functions
+
+- Create a file named `auth.ts` in the `src/lib/` directory.
+
+In `auth.ts` add the following code to create the authentication functions:
+
+```ts title="src/lib/auth.ts"
+import { supabase } from "./supabaseClient";
+
+export async function signUp(email: string, password: string) {
+    return await supabase.auth.signUp({
+        email,
+        password,
+        options: {
+          emailRedirectTo: 'http://localhost:3000/profile'
+        }
+    })
+}
+
+export async function signIn(email: string, password: string) {
+    return await supabase.auth.signInWithPassword({
+        email,
+        password
+    })
+}
+
+export async function signInWithGithub() {
+    return await supabase.auth.signInWithOAuth({
+        provider: 'github',
+        options: {
+          redirectTo: 'http://localhost:3000/profile'
+        }
+    });
+}
+
+export async function getSessionData() {
+    return await supabase.auth.getSession();
+}
+
+export async function setSessionData(access_token: string, refresh_token: string) {
+    return await supabase.auth.setSession({
+        access_token,
+        refresh_token,
+    })
+}
+
+export function AuthState() {
+    return supabase.auth.onAuthStateChange((event, session) => {
+        console.log(event, session);
+    })
+}
+
+export async function signOut() {
+    return await supabase.auth.signOut()
+}
+```
+
+:::tip
+Check [Supabase User Management](https://supabase.com/dashboard/project/qiilhzulchhvmptaayfz/api?page=users) to learn more about these functions.
```

Comment:
I think this link is for your specific project, I don't think it will work for everyone.

Reply:
> I think this link is for your specific project, I don't think it will work for everyone.

Fixed it. Thank you :heart: